cypress/e2e/awx/resources/hosts.cy.ts

```diff
@@ -37,89 +37,105 @@ describe('host and inventory host', () => {
-  }
+    cy.getByDataCy('variables').type('test: true');
+    cy.clickButton(/^Create host/);
+    cy.hasDetail(/^Name$/, hostName);
+    cy.hasDetail(/^Description$/, 'This is the description');
+    cy.hasDetail(/^Variables$/, 'test: true');
 
-  cy.getByDataCy('variables').type('test: true');
-  cy.clickButton(/^Create host/);
-  cy.hasDetail(/^Name$/, hostName);
-  cy.hasDetail(/^Description$/, 'This is the description');
-  cy.hasDetail(/^Variables$/, 'test: true');
-
-  return hostName;
-}
-
-function editHost(inventoryID: number, inventory_host: boolean, hostName: string) {
-  if (inventory_host) {
-    cy.visit(
-      `/infrastructure/inventories/inventory/${inventoryID}/hosts/?page=1&perPage=10&sort=name`
-    );
-  } else {
-    cy.visit('/infrastructure/hosts?page=1&perPage=10&sort=name');
-  }
-  cy.searchAndDisplayResource(hostName);
-
-  cy.get(`[data-cy='edit-host']`).click();
-  cy.verifyPageTitle('Edit host');
-  cy.get('[data-cy="description"]').clear().type('This is the description edited');
-  cy.get(`[data-cy='Submit']`).click();
-
-  cy.hasDetail(/^Description$/, 'This is the description edited');
-}
-
-function deleteHost(inventoryID: number, inventory_host: boolean, hostName: string) {
-  if (inventory_host) {
-    cy.visit(
-      `/infrastructure/inventories/inventory/${inventoryID}/hosts/?page=1&perPage=10&sort=name`
-    );
-  } else {
-    cy.visit('/infrastructure/hosts?page=1&perPage=10&sort=name');
+    return hostName;
   }
-  cy.contains(hostName).click();
-  cy.selectDetailsPageKebabAction('delete-host');
-
-  if (inventory_host) {
-    cy.get('[data-cy="empty-state-title"]').contains(
-      /^There are currently no hosts added to this inventory./
-    );
-  } else {
+
+  function editHost(inventoryID: number, inventory_host: boolean, hostName: string) {
+    if (inventory_host) {
+      cy.visit(
+        `/infrastructure/inventories/inventory/${inventoryID}/hosts/?page=1&perPage=10&sort=name`
+      );
+    } else {
+      cy.visit('/infrastructure/hosts?page=1&perPage=10&sort=name');
+    }
     cy.searchAndDisplayResource(hostName);
-    cy.contains('No results found');
+
+    cy.get(`[data-cy='edit-host']`).click();
+    cy.verifyPageTitle('Edit host');
+    cy.get('[data-cy="description"]').clear().type('This is the description edited');
+    cy.get(`[data-cy='Submit']`).click();
+
+    cy.hasDetail(/^Description$/, 'This is the description edited');
   }
-}
 
-function createAndEditAndDeleteHost(inventory_host: boolean, inventory: Inventory) {
-  //create
-  const hostName = createAndCheckHost(inventory_host, inventory.name);
+  function deleteHost(inventoryID: number, inventory_host: boolean, hostName: string) {
+    if (inventory_host) {
+      cy.visit(
+        `/infrastructure/inventories/inventory/${inventoryID}/hosts/?page=1&perPage=10&sort=name`
+      );
+    } else {
+      cy.visit('/infrastructure/hosts?page=1&perPage=10&sort=name');
+    }
+    cy.contains(hostName).click();
```

Comment:
There is missing cy.searchAndDisplayResource(hostName);

It must be done first before doing cy.contains(hostName).click();

otherwise with more hosts in the database, you may not see that host in the page and test may fall.

Reply:
this was added